### ISSUE TITLE
Export completely overwrites directories

### ISSUE THREAD

**Issue by jscheel** [godotengine/godot#36034]
**Godot version:**
3.2

**OS/device including version:**
macOS 10.15.2

**Issue description:**
I accidentally exported an iOS project out to the same directory my game project was in. Because the names were the same, it overwrote my entire game directory without warning.

**Steps to reproduce:**
Export iOS project to parent directory containing game project, save as same name. Instead of throwing a warning up or just adding files to the directory, it overwrites everything.

**Comment by PabloNeirotti** [comment]
Was gonna report the same. Very sad to loose everything. Copy-pasting below so I don't open the same issue.

**Godot version:**
v3.2.stable.oficial

**OS/device including version:**
macOS Catalina 10.15.2 (19C57)
MacBook Pro (15-inch, 2019)

**Issue description:**
When exporting to iOS, if the target folder is the parent of the current project's folder, the contents of the folder will be replaced with iOS stuff, and your project code, assets, sweat and tears will be gone. Well, not the latter.

**Minimal reproduction project:**
It's all gone.

**Comment by akien-mga** [comment]
CC @godotengine/ios

**Comment by bruvzg** [comment]
https://github.com/godotengine/godot/blob/ef5891091bceef2800b4fae4cd85af219e791467/platform/iphone/export/export.cpp#L1007-L1012

It's deleting `binary_name.xcodeproj` and `binary_name` in the target folder.

I'm not sure what's the best way to handle this, but at least it should be changed to `move_to_trash`.

**Comment by PabloNeirotti** [comment]
All I know is that it deletes the whole game project, code and assets.

It should not delete anything unless it either provides a proper warning for permanent deletion of “folder name” like any other app, or do it only if it is confident enough it’s overwriting an old export, not a Godot project or something else just because names match.

**Comment by neikeq** [comment]
It should not be allowed to be exported to the project folder unless it's an empty sub-directory. A similar thing happened with the 2.1 to 3.0 migration tool. In addition it should warn that it's about to remove all the target directory's contents and ask for confirmation.
_(Sorry, self-assigned by accident from my phone)_

**Comment by akien-mga** [comment]
> It should not be allowed to be exported to the project folder unless it's an empty sub-directory. A similar thing happened with the 2.1 to 3.0 migration tool. In addition it should warn that it's about to remove all the target directory's contents and ask for confirmation.

I agree with both. It could also validate that the contents match a typical iOS export folder before erasing.
